stage1/usr_from_kvm/kernel.mk

```diff
@@ -0,0 +1,78 @@
+# custom kernel compilation
+KERNEL_VERSION := 4.1.1
+KERNEL_TMP := $(BUILDDIR)/tmp/usr_from_kvm/kernel
+KERNEL_NAME := linux-$(KERNEL_VERSION)
+KERNEL_TARBALL := $(KERNEL_NAME).tar.xz
+KERNEL_TARGET_FILE := $(KERNEL_TMP)/$(KERNEL_TARBALL)
+KERNEL_SRCDIR := $(KERNEL_TMP)/$(KERNEL_NAME)
+KERNEL_BUILDDIR := $(abspath $(KERNEL_TMP)/build-$(KERNEL_VERSION))
+KERNEL_URL := https://www.kernel.org/pub/linux/kernel/v4.x/$(KERNEL_TARBALL)
+KERNEL_MAKEFILE := $(KERNEL_SRCDIR)/Makefile
+KERNEL_STUFFDIR := $(MK_SRCDIR)/kernel
+KERNEL_SRC_CONFIG := $(KERNEL_STUFFDIR)/cutdown-config
+KERNEL_PATCHES := $(abspath $(KERNEL_STUFFDIR)/patches/*.patch)
+KERNEL_BUILD_CONFIG := $(KERNEL_BUILDDIR)/.config
+KERNEL_BZIMAGE := $(KERNEL_BUILDDIR)/arch/x86/boot/bzImage
+KERNEL_ACI_BZIMAGE := $(ACIROOTFSDIR)/bzImage
+
+$(call setup-stamp-file,KERNEL_STAMP,/build_kernel)
+$(call setup-stamp-file,KERNEL_PATCH_STAMP,/patch_kernel)
+$(call setup-dep-file,KERNEL_PATCHES_DEPMK)
+
+CREATE_DIRS += $(KERNEL_TMP) $(KERNEL_BUILDDIR)
+INSTALL_FILES += \
+	$(KERNEL_SRC_CONFIG):$(KERNEL_BUILD_CONFIG):- \
+	$(KERNEL_BZIMAGE):$(KERNEL_ACI_BZIMAGE):-
+UFK_STAMPS += $(KERNEL_STAMP)
+
+$(KERNEL_STAMP): $(KERNEL_ACI_BZIMAGE)
+	touch "$@"
+
+$(KERNEL_BZIMAGE): KERNEL_SRCDIR := $(KERNEL_SRCDIR)
+$(KERNEL_BZIMAGE): KERNEL_BUILDDIR := $(KERNEL_BUILDDIR)
+$(KERNEL_BZIMAGE): $(KERNEL_BUILD_CONFIG) $(KERNEL_PATCH_STAMP)
+	$(MAKE) -C "$(KERNEL_SRCDIR)" O="$(KERNEL_BUILDDIR)" bzImage
+
+-include $(KERNEL_PATCHES_DEPMK)
+$(KERNEL_PATCH_STAMP): PERL := $(PERL)
+$(KERNEL_PATCH_STAMP): DEPSGENTOOL := $(DEPSGENTOOL)
+$(KERNEL_PATCH_STAMP): KERNEL_PATCHES := $(KERNEL_PATCHES)
+$(KERNEL_PATCH_STAMP): KERNEL_PATCHES_DEPMK := $(KERNEL_PATCHES_DEPMK)
+$(KERNEL_PATCH_STAMP): KERNEL_SRCDIR := $(KERNEL_SRCDIR)
+$(KERNEL_PATCH_STAMP): $(KERNEL_MAKEFILE)
+	set -e; \
+	shopt -s nullglob; \
+	"$(PERL)" "$(DEPSGENTOOL)" --target '$$(KERNEL_MAKEFILE)' --suffix=.patch $(KERNEL_PATCHES) >"$(KERNEL_PATCHES_DEPMK)"; \
```

Comment:
If #1155 is merged before this PR, then this line will have to be replaced with:
`$(GO_ENV) "$(DEPSGENTOOL)" glob --target '$$(KERNEL_MAKEFILE)' --suffix=.patch $(KERNEL_PATCHES) >"$(KERNEL_PATCHES_DEPMK)"; \`